definition/IServerEvent.ts

```diff
@@ -0,0 +1,14 @@
+import { IUser } from './IUser';
+
+export enum IServerEventType {
+	FAILED_LOGIN_ATTEMPT = 'failed-login-attempt',
+	LOGIN = 'login',
+}
+
+export interface IServerEvent {
+	_id: string;
+	t: IServerEventType;
+	ts: Date;
+	ip?: string;
+	u?: IUser;
```

Comment:
from what I saw both `ip` and `u` are always set. is there a special reason to make them optional?
```suggestion
	ip: string;
	u: IUser;
```